hack/licenser/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+var (
+	depFile = flag.String("f", "go.mod", "File with the list of dependencies")
+	dir     = flag.String("d", "", "Project directory")
+)
+
+type Dependency struct {
+	Name    string
+	Version string
+	License string
+}
+
+type Dependencies struct {
+	List []*Dependency
+}
+
+func main() {
+	flag.Parse()
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+
+	deps, err := loadFile(filepath.Join(*dir, *depFile))
```

Comment:
I wonder if parsing the output of `go list -deps -test -json` [1] might be better than parsing `go.mod` itself because the former is is less likely to change with newer Go versions.

[1] probably needs to be tweaked a bit to get the desired output

Reply:
I tried to get output from `go list` but wasn't been able to get something useful from it. Instead I decided to use `modules.txt` file which is generated when `go mod vendor` is executed. It will produce easy to parse text file like:
```
# github.com/davecgh/go-spew v1.1.1
github.com/davecgh/go-spew/spew
# github.com/docker/spdystream v0.0.0-20181023171402-6480d4af844c
github.com/docker/spdystream
github.com/docker/spdystream/spdy
....
```

Reply:
I had some luck parsing the output of `go list` and running licence-detector. Happy to collaborate on this if you are up for it.